dot/core/service.go

```diff
@@ -373,8 +373,11 @@ func (s *Service) handleChainReorg(prev, curr common.Hash) error {
 			externalExt := make(types.Extrinsic, 0, 1+len(ext))
 			externalExt = append(externalExt, byte(types.TxnExternal))
 			externalExt = append(externalExt, ext...)
-			txv, err := rt.ValidateTransaction(externalExt)
-			if err != nil {
+			txv, txnValidityErr, err := rt.ValidateTransaction(externalExt)
+			if err != nil || txnValidityErr != nil {
+				if txnValidityErr != nil {
+					err = txnValidityErr.Error()
+				}
```

Comment:
Since we always replace `err` if `txnValidityErr` is not nil so I think we can avoid nested if's using:
```suggestion
if txnValidityErr != nil {
  err = txnValidityErr.Error()
}

if err != nil {
  logger.Debugf("failed to validate transaction for extrinsic %s: %s", ext, err)
  continue
}
```